### PR TITLE
configure: Add a new check for F08 support

### PR DESCRIPTION
**Pull request by yfguo**
The new check is for the case where a async subarray is passed to a
assumed-type assumed-rank async dummy argument of a interface in a
module. ifort 18 is known to give compile error on it.

### CODE REVIEW

**Comment by yfguo** [comment]
test:jenkins/ch3/tcp

**Comment by yfguo** [comment]
@raffenet The fix seems successfully disables F08 on machines with `ifort` v18 (mac nodes). This PR should be succeeded by pmodels/confdb#1

File: confdb/aclocal_fc.m4 [hunk 1]
```diff
@@ -384,7 +384,6 @@ PAC_FC_MODULE_EXT
 PAC_FC_MODULE_INCFLAG
 PAC_FC_MODULE_OUTFLAG
 ])
-dnl
```
Comment (pavanbalaji) [inline confdb/aclocal_fc.m4]: Unnecessary change, should be removed from this patch.

**Comment by pavanbalaji** [comment]
@yfguo if this change is being added to the pmodels/confdb, then you need a patch here to update the git hash that pulls `confdb`.

**Comment by yfguo** [comment]
test:jenkins/ch3/tcp

**Comment by yfguo** [comment]
test:jenkins/ch3/tcp

**Comment by raffenet** [comment]
@yfguo how do earlier Intel compilers treat your test program? Will they pass it erroneously? Just trying to figure out what to expect.

**Comment by yfguo** [comment]
The earlier versions (v16 and v17) of `ifort` do not have the compile error as `ifort` v18 does. So in that sense, the check is passed and F08 bindings will be built. Judging from the results from the MPICH test suite, I think the v16 and v17 version compiles probably handles the code correctly. My guess is this might be a problem introduced as the v18 trying to fully support F08 and some of the F15 features.

**Comment by raffenet** [comment]
K. My manual tests look good with this change. I don't think we need to wait for the rest of the tests to complete, since we're only looking at one specific compiler not in the review job. I will signoff and merge.

**Comment by yfguo** [comment]
I agree
